applications/GeoMechanicsApplication/custom_utilities/check_utilities.h

```diff
@@ -27,5 +30,10 @@ class KRATOS_API(GEO_MECHANICS_APPLICATION) CheckUtilities
     static void CheckDomainSize(double                            DomainSize,
                                 std::size_t                       Id,
                                 const std::optional<std::string>& PrintName = std::nullopt);
+								
+	static void CheckNodalVariables(const Geometry<Node>&            rGeometry,
+                                    const std::vector<VariableData>& rVariables);
```

Comment:
To pass collections of variables, we would normally use `Geo::ConstVariableDataRefs` (defined in `geo_aliases.h`) rather than `std::vector<VariableData>`. In other words, to align with the rest of the code we may want to change the function parameter type to `const Geo::ConstVariableDataRefs&`:
```suggestion
                                    const Geo::ConstVariableDataRefs& rVariableRefs);
```
To construct an a suitable object of this type, you need to use `std::cref`. I'll add an example to your test code.

This suggestions also applies to `CheckNodalDof`.

I expect that by applying these changes you will get rid of the slicing issues reporting by SonarQube as well as Clang-Tidy.

One more thing, what do you think about trying to make the function's name a little bit more specific, something like `CheckHasNodalSolutionStepData`?